java/java.editor/src/org/netbeans/modules/java/editor/javadoc/JavadocCompletionCollector.java

```diff
@@ -0,0 +1,293 @@
+import org.netbeans.api.editor.mimelookup.MimeRegistration;
+import org.netbeans.api.java.source.CompilationController;
+import org.netbeans.api.java.source.CompilationInfo;
+import org.netbeans.api.java.source.ElementHandle;
+import org.netbeans.api.java.source.JavaSource;
+import org.netbeans.api.java.source.support.ReferencesCount;
+import org.netbeans.api.lsp.Completion;
+import org.netbeans.api.lsp.TextEdit;
+import org.netbeans.modules.editor.java.JavaCompletionCollector;
+import org.netbeans.modules.editor.java.Utilities;
+import org.netbeans.modules.java.editor.base.javadoc.JavadocCompletionUtils;
+import org.netbeans.modules.parsing.api.ParserManager;
+import org.netbeans.modules.parsing.api.ResultIterator;
+import org.netbeans.modules.parsing.api.Source;
+import org.netbeans.modules.parsing.api.UserTask;
+import org.netbeans.modules.parsing.spi.ParseException;
+import org.netbeans.spi.lsp.CompletionCollector;
+import org.openide.util.Exceptions;
+
+/**
+ *
+ * @author Dusan Balek
+ */
+@MimeRegistration(mimeType = "text/x-java", service = CompletionCollector.class)
+public class JavadocCompletionCollector implements CompletionCollector {
+
+    @Override
+    public boolean collectCompletions(Document doc, int offset, Completion.Context context, Consumer<Completion> consumer) {
+        AtomicBoolean ret = new AtomicBoolean(true);
+        if (JavadocCompletionUtils.isJavadocContext(doc, offset)) {
+            try {
+                ParserManager.parse(Collections.singletonList(Source.create(doc)), new UserTask() {
+                    @Override
+                    public void run(ResultIterator resultIterator) throws Exception {
+                        CompilationController controller = CompilationController.get(resultIterator.getParserResult(offset));
+                        controller.toPhase(JavaSource.Phase.ELEMENTS_RESOLVED);
+                        JavadocCompletionTask<Completion> task = JavadocCompletionTask.create(offset, new ItemFactoryImpl(controller, offset),
+                                context.getTriggerKind() == Completion.TriggerKind.TriggerForIncompleteCompletions, () -> false);
+                        task.run(resultIterator);
+                        List<Completion> results = task.getResults();
+                        if (results != null) {
+                            for (Iterator<Completion> it = results.iterator(); it.hasNext();) {
+                                Completion item = it.next();
+                                if (item == null) {
+                                    it.remove();
+                                }
+                            }
+                            results.forEach(consumer);
+                        }
+                        if (task.hasAdditionalItems()) {
+                            ret.set(false);
+                        }
+                    }
+                });
+            } catch (ParseException ex) {
+                Exceptions.printStackTrace(ex);
```

Comment:
As the completion result  set is usually collected form multiple collector instances, it is IMHO better (in case of failure of one of them) just to log the exception and show a partial result set collected from the other collectors (then notify users about the exception and show nothing).